### ISSUE TITLE
Fix PNG rendering

### ISSUE THREAD

**Issue by chubin** [chubin/wttr.in#919]
PNG rendering is temporary disabled.

```
$ curl wttr.in/Nuremberg.png
PNG support temporary disabled
```

**Comment by edi9999** [comment]
Same issue for me :)

**Comment by truerico** [comment]
same issue here too

**Comment by lecoxsportif** [comment]
same

**Comment by morfeus2** [comment]
Any idea why was it disabled?

**Comment by Intellek-help** [comment]
Can't find any info on why it has been disabled, is this only the png image or emojis too?

**Comment by edi9999** [comment]
Here is what I use now as a replacement : 

```
img="$(mktemp /tmp/weather_XXXX.png)"
textimg -i "$(curl --silent "wttr.in/$town?q")" -F 30 -o "$img"
ristretto "$img"
```

textimg can be installed here : `go install github.com/jiro4989/textimg/v3@latest`

**Comment by liam8888999** [comment]
is there any update on this

**Comment by chubin** [comment]
I started porting the existing Python implementation to Go (the Python implementation is too slow for the current load level). Let me know if you want to help

**Comment by nirvam** [comment]
> I started porting the existing Python implementation to Go (the Python implementation is too slow for the current load level). Let me know if you want to help

@chubin edi9999 metioned https://github.com/jiro4989/textimg in his workaround, which seems a good module for ANSI text to PNG conversion.

**Comment by chubin** [comment]
I've skimmed through it, and it seems to be pretty fit for our needs. It has emoji support, wide runes, true colors, etc. I will play with it, and check how easy it could be integrated

**Comment by edi9999** [comment]
The only bug I had so far was when there was a "thunderstorm alert". In the terminal, it shows as an animated "thunder" icon.
In that case, the PNG output had some strange chafracters with textimg.

But that was the only issue I had in several weeks of usage.

**Comment by morfeus2** [comment]
> I started porting the existing Python implementation to Go (the Python implementation is too slow for the current load level). Let me know if you want to help

Now I understand, good to know the motivation.
Unfortunately I don't know Go at all, otherwise I would have offered my help.

**Comment by ivan-ristovic** [comment]
> @chubin edi9999 metioned https://github.com/jiro4989/textimg in his workaround, which seems a good module for ANSI text to PNG conversion.

> I've skimmed through it, and it seems to be pretty fit for our needs. It has emoji support, wide runes, true colors, etc. I will play with it, and check how easy it could be integrated

Seems great and efficient for v1 output. I couldn't make it work with the v2 output though (only the weather icons worked well but not the graph itself). If anyone else managed to make it work, please share.

**Comment by edi9999** [comment]
Yes indeed, with the v2 output the textimg software doesn't give the right output ...

**Comment by edi9999** [comment]
For example, the thunderstorm output is buggy with the textimg utility : 

![image](https://github.com/chubin/wttr.in/assets/2071336/8c3611c1-0866-45ee-939a-dce562d309d2)

**Comment by sholl** [comment]
Is there any progres on this issue?

**Comment by vitich** [comment]
Just install phantomjs and imagemagick and 

${voffset 65}
${texeci 30 phantomjs $HOME/phantomjs/rasterize.js http://wttr.in/Kyiv /dev/stdout | convert - -crop 200x140+0+0 -transparent black /tmp/wttr.png}
${image /tmp/wttr.png -p}

![Screenshot_2024-09-06_06-29-14](https://github.com/user-attachments/assets/6e116826-0a5f-4df6-863b-d01df2040495)

**Comment by Bezrazniti** [comment]
PNG support temporary disabled

Haven't you solved the problem yet? Or can we say goodbye to PNG?

**Comment by vitich** [comment]
> PNG support temporary disabled
> 
> Haven't you solved the problem yet ? Or can we say goodbye to PNG ? :(((

Look at my comment above 

![Image](https://github.com/user-attachments/assets/87ad7a3d-2e58-422c-bd60-23009c6b3ba4)

**Comment by chubin** [comment]
PNG rendering will be reenabled soon. Thank you for your patience and loyality to our service

**Comment by chubin** [comment]
PNG support is back. Please test it, and you notice anything suspicious, let me know.

**Comment by Bezrazniti** [comment]
> PNG support is back. Please test it, and you notice anything suspicious, let me know.

Thank you! Everything works on Android and everything is fine!
